runtime/commonMain/src/kotlinx/serialization/internal/SerializationConstructorMarkerCompatibility.kt

```diff
@@ -8,5 +8,5 @@ package kotlinx.serialization
 
 // Binary compatibility with the plugin
 @Suppress("UNUSED")
-@Deprecated("Inserted into generated code and should not be used directly", level = DeprecationLevel.HIDDEN)
+@Deprecated("Inserted into generated code and should not be used directly", level = DeprecationLevel.HIDDEN) // TODO still used by the plugin
```

Comment:
I do not remember any plans to delete it. It should be used by the plugin, just be invisible to users, hence HIDDEN.

Reply:
We have two `SerializationConstructorMarker`, one in `kotlinx.serialization` package and another one in `kotlinx.serialization.internal`.
This is the former and it should be removed at some point